overlord/configstate/configcore/ctrlaltdel.go

```diff
@@ -0,0 +1,142 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/sysconfig"
+	"github.com/snapcore/snapd/systemd"
+)
+
+// Systemd unit name to control ctrl-alt-del behaviour
+const (
+	ctrlAltDelTarget = "ctrl-alt-del.target"
+)
+
+// Supported actions for core.system.ctrl-alt-del-action
+const (
+	ctrlAltDelReboot = "reboot"
+	ctrlAltDelNone   = "none"
+)
+
+func init() {
+	supportedConfigurations["core.system.ctrl-alt-del-action"] = true
+}
+
+type sysdCtrlAltDelLogger struct{}
+
+func (l *sysdCtrlAltDelLogger) Notify(status string) {
+	logger.Noticef("CtrlAltDel handler: %s", status)
+}
+
+// switchCtrlAltDelAction configures the systemd handling of the special
+// ctrl-alt-del keyboard sequence. This function supports configuring
+// systemd to trigger a reboot, or ignore the key sequence.
+func switchCtrlAltDelAction(action string, opts *fsOnlyContext) error {
+	validAction := map[string]bool{
+		ctrlAltDelReboot: true,
+		ctrlAltDelNone:   true,
+	}
+	if !validAction[action] {
```

Comment:
or since there are only 2 actions possible we can do this and skip the map
```suggestion
	if validAction != "reboot" && validAction != "none" {
```